assets/text-with-image.js

```diff
@@ -0,0 +1,61 @@
+class TextWithImageSpacer {
+  constructor(section) {
+    this.section = section;
+
+    this.classes = {
+      spacer: "text-image__spacer",
+      one: "palette-one",
+      two: "palette-two",
+      three: "palette-three",
+    }
+  }
+
+  init() {
+    if (!this.section) return false;
+
+    this.setColor();
+  }
+
+  setColor() {
+    const children = [...this.section.children],
+          classes = [this.classes.one, this.classes.two, this.classes.three];
+
+    children.forEach(child => {
+      if (!child.classList.contains(this.classes.spacer)) return false;
+
+      const prevSection = this.section.previousElementSibling;
+
+      if (!prevSection) return false;
+
+      const prevChildren = [...prevSection.children];
+
+      prevChildren.forEach(item => {
+        const itemClasses = classes.filter(className => item.classList.contains(className)),
+              childClasses = classes.filter(className => child.classList.contains(className));
+
+        itemClasses.forEach(itemClass => {
+          const replaceClass = childClasses.find(childClass => child.classList.contains(childClass));
+
+          if (replaceClass) {
+            child.classList.replace(replaceClass, itemClass);
+          }
+        })
+      })
+    })
+  }
+}
```

Comment:
WDYT about functional approach?

```javascript
const setColor = (section) => {
  const SPACER = "text-image__spacer",
        CLASSES = ["palette-one", "palette-two", "palette-three"]

  const children = [...section.children]

  children.forEach(child => {
    if (!child.classList.contains(SPACER)) return

    const prevSection = section.previousElementSibling

    if (!prevSection) return

    const prevChildren = [...prevSection.children]

    prevChildren.forEach(item => {
      const itemClasses = CLASSES.filter(className => item.classList.contains(className)),
            childClasses = CLASSES.filter(className => child.classList.contains(className))

      itemClasses.forEach(itemClass => {
        const replaceClass = childClasses.find(childClass => child.classList.contains(childClass))

        if (replaceClass) {
          child.classList.replace(replaceClass, itemClass)
        }
      })
    })
  })
}

const textWithImageSpacer = (section) => {
  if (!this.section) return false

  setColor(section)
}
```

Reply:
This approach is good as well, but I just try to stick to 1 style of code that has already been used in other JS files

Reply:
"One style to rule them all" ;-)

Reply:
![image](https://github.com/booqable/tough-theme/assets/1877965/4abc030c-9c10-42d1-b5ac-ea7e87d38032)